File: Problem_3_BruteForce.java
// Time complexity - O(n^2)
// Space complexity - O(1)
// Brute force approach using 2 loops and checking each element.

class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        if(matrix == null || matrix.length == 0){
            return false;
        }
        int n = matrix.length; int m = matrix[0].length;
        for(int  i =0; i<n; i++){
            for(int j = 0 ; j< m ; j++){
                if(matrix[i][j] == target){
                    return true;
                }
            }
        }
        return false;
    }
}

File: problem3_bruteForce.py
"""
Time Complexity : O(m*n)
Space Complexity : O(1) No new data structure is used
Did this code successfully run on Leetcode :
Three line explanation of solution in plain english:
Iterate through the elements and return the target if found

// Your code here along with comments explaining your approach
"""
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        rows = len(matrix)
        cols = len(matrix[0])
        
        for row in range(rows):
            for col in range(cols):
                if matrix[row][col] == target:
                    return True
        return False